cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/DatasetBasedTimeScheduleStore.java

```diff
@@ -95,7 +95,12 @@ public void storeTrigger(OperableTrigger newTrigger, boolean replaceExisting) th
   public void storeJobsAndTriggers(Map<JobDetail, Set<? extends Trigger>> triggersAndJobs,
                                    boolean replace) throws JobPersistenceException {
     super.storeJobsAndTriggers(triggersAndJobs, replace);
-    // TODO (sree): Add persisting job and triggers
+    for (Map.Entry<JobDetail, Set<? extends Trigger>> e: triggersAndJobs.entrySet()) {
+      storeJob(e.getKey(), replace);
+      for (Trigger trigger: e.getValue()) {
+        storeTrigger((OperableTrigger) trigger, replace);
+      }
+    }
```

Comment:
That will cause the storeJob to be called again and again for every storeTrigger which is unnecessary work as the job is same for all its associated trigger and since storeJobAndTrigger just makes call to other methods which we are doing directly here I don't think it is a code duplication issue.